pkg/controllers/session/sessioncontroller.go

```diff
@@ -145,6 +145,30 @@ func (s *SessionController) reconcileSession(ssName string) error {
 
 	timeUntilExpires := expires.Sub(now)
 
+	// clean up old (3 hours later) sessions, and only if they are finished
+	if expires.Add(time.Hour*3).Before(now) && ss.Status.Finished {
+		// first we need to delete the vmclaims
+		err := s.hfClientSet.HobbyfarmV1().VirtualMachineClaims().DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("hobbyfarm.io/session=%s", ss.Name),
+		})
+
+		if err != nil {
```

Comment:
Is this going to error out if there are no vmclaims found with the corresponding label?

Reply:
In my testing, no. I created a session with no corresponding vm claims, and the session deleted without an error.